src/lib.rs

```diff
@@ -40,12 +39,48 @@ pub struct Command {
     env: HashMap<String, String>,
 }
 
-impl Command {
-    pub fn new<S, C, ArgType, Cmds>(
-        name: S,
-        command: C,
-        args: Cmds,
-    ) -> Result<Command, CommandError>
+impl AsRef<Command> for Command {
+    fn as_ref(&self) -> &Command {
+        self
+    }
+}
+
+pub trait CommandOperations
+where
+    Self: Sized,
+{
+    fn new<S, C, ArgType, Cmds>(name: S, command: C, args: Cmds) -> Result<Self, CommandError>
+    where
+        S: AsRef<str>,
+        C: AsRef<str>,
+        ArgType: AsRef<str>,
+        Cmds: IntoIterator<Item = ArgType>;
+
+    fn full_cmd<S, C>(name: S, command_string: C) -> Result<Self, CommandError>
+    where
+        S: AsRef<str>,
+        C: AsRef<str>;
+
+    fn cur_dir<D>(&mut self, dir: D) -> &mut Self
+    where
+        D: AsRef<Path>;
```

Comment:
This is just an idea/proposal that I read about a week ago. When you use this value, you need a `PathBuf`. So, rather than requesting a reference and then cloning it, you could request a type that can be converted into `PathBuf`. Many types in the standard library implement `Into`, including the usual suspects one would most typically use here.